packages/staking/src/features/store/delegationPortfolioStore/stateMachine/normalizePercentages.ts

```diff
@@ -1,19 +1,36 @@
 import { PERCENTAGE_SCALE_MAX } from '../constants';
+import { sumPercentagesLossless } from './sumPercentagesLossless';
 
-// Rounds values of integer properties of an input array of objects, while keeping the sum of the numbers equal to 100 (PERCENTAGE_SCALE_MAX)
-// Requires input array to sum to 100
-// Examples: normalizePercentages([{val: 33.33},{val: 33.33},{val: 33.33}], 'val')
-//                            =>  [{val: 34},   {val: 33},   {val: 33}   ]
-export const normalizePercentages = <K extends string, T extends { [key in K]: number }>(items: T[], key: K) => {
-  const currentSum = items.reduce((acc, item) => acc + item[key], 0);
+// Rounds values of properties of an input array of objects, while keeping the sum of the numbers equal to 100 (PERCENTAGE_SCALE_MAX)
+// Requires input array to sum to 100 +- epsilon
+// Works at arbitrary decimal precision
+// Examples: normalizePercentages({items: [{val: 33.33},{val: 33.33},{val: 33.33}], key: 'val', decimals: 0})
+//                            =>          [{val: 34},   {val: 33},   {val: 33}   ]
+//           normalizePercentages({items: [{val: 33.333},{val: 33.333},{val: 33.333}], key: 'val', decimals: 2})
+//                            =>          [{val: 33.34}, {val: 33.33}, {val: 33.33} ]
```

Comment:
I'd make this a proper docstring for better development experience, so that it's shown by editor intellisense when hovering over the function invocation in the code
```suggestion
/**
 * Rounds values of properties of an input array of objects, while keeping the sum of the numbers equal to 100 (PERCENTAGE_SCALE_MAX)
 * Requires input array to sum to 100 +- epsilon
 * Works at arbitrary decimal precision
 * Examples:
 * ```
 *           normalizePercentages({items: [{val: 33.33},{val: 33.33},{val: 33.33}], key: 'val', decimals: 0})
 *                             =>          [{val: 34},   {val: 33},   {val: 33}   ]
 * ```
 * ```
 *           normalizePercentages({items: [{val: 33.333},{val: 33.333},{val: 33.333}], key: 'val', decimals: 2})
 *                            =>          [{val: 33.34}, {val: 33.33}, {val: 33.33} ]
 * ```
 */
```